internal/uvm/network.go

```diff
@@ -23,6 +26,43 @@ var (
 	ErrNetNSNotFound = errors.New("network namespace not found")
 )
 
+// NetworkEndpoint is a struct containing the endpoint id of a network namespace
+type NetworkEndpoint struct {
+	EndpointID string
+}
+
+// Close frees the network endpoints resources
+func (endpoint *NetworkEndpoint) Close(ctx context.Context, namespace string) error {
+	err := hns.RemoveNamespaceEndpoint(namespace, endpoint.EndpointID)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		log.G(ctx).WithFields(logrus.Fields{
+			"endpointID": endpoint,
+			"netID":      namespace,
+		}).Warn("removing endpoint from namespace: does not exist")
+	}
+	return nil
+}
+
+// CloseAllNetworkEndpoints frees the resources for all of the network endpoints
+func CloseAllNetworkEndpoints(ctx context.Context, namespace string, endpoints []*NetworkEndpoint) error {
```

Comment:
I think it might be more consistent to, instead of storing an array of pointers to NetworkEndpoint structs, doing something like what you've done with the layers, where we store a pointer to a slice of strings that represent the endpoint IDs. What do you think?

Reply:
Yea this was kind of an afterthought from the first version of this pr because I was thinking of how to handle network cleanup in general. I like that idea though

Reply:
This might not be in this change. Might be better to redo all the network/layers stuff in another pr